trust/token.c

```diff
@@ -272,11 +283,25 @@ loader_load_directory (p11_token *token,
 		return 0;
 	}
 
+	paths = p11_array_new (NULL);
+
 	while ((dp = readdir (dir)) != NULL) {
 		path = p11_path_build (directory, dp->d_name, NULL);
 		return_val_if_fail (path != NULL, -1);
 
-		ret = loader_load_if_file (token, path);
+		if (!p11_array_push (paths, path)) {
+			free (path);
+			return -1;
+		}
+	}
+
+    closedir(dir);
```

Comment:
Formatting needs fixing. You are mixing TABs and spaces + put space between function name and brackets.